### ISSUE TITLE
Dynamic Serilog Provider does not respect manual configuration

### ISSUE THREAD

**Issue by jpenniman** [SteeltoeOSS/Steeltoe#478]
## Describe the bug
When hardcoding the Serilog configuration, the levels are over written by the default values in `SerlilogOptions`.

``` cs
public static ILoggingBuilder AddDynamicSerilog(this ILoggingBuilder builder, LoggerConfiguration loggerConfiguration, bool preserveStaticLogger = false)
        {
            if (builder.Services.Any(sd => sd.ServiceType == typeof(IDynamicLoggerProvider)))
            {
                throw new InvalidOperationException("An IDynamicLoggerProvider has already been configured! Call 'AddDynamicSerilog' earlier in program.cs (before adding Actuators) or remove duplicate IDynamicLoggerProvider entries.");
            }

            var serilogOptions = new SerilogOptions(builder.Services.BuildServiceProvider().GetRequiredService<IConfiguration>());
            loggerConfiguration ??= new LoggerConfiguration();

            // Add a level switch that controls the "Default" level at the root
            var levelSwitch = new LoggingLevelSwitch(serilogOptions.MinimumLevel.Default);  <=== BUG
            loggerConfiguration.MinimumLevel.ControlledBy(levelSwitch); <=== BUG
            var logger = loggerConfiguration.CreateLogger();
            if (!preserveStaticLogger)
            {
                Log.Logger = logger;
            }

            builder.Services.AddSingleton<ISerilogOptions>(serilogOptions);
            builder.Services.AddSingleton(levelSwitch);
            builder.Services.AddSingleton(logger);
            builder.Services.TryAddEnumerable(ServiceDescriptor.Singleton<ILoggerProvider, SerilogDynamicProvider>());
            builder.Services.AddSingleton((p) => p.GetServices<ILoggerProvider>().OfType<IDynamicLoggerProvider>().SingleOrDefault());
            return builder;
        }
```

## Steps to reproduce 
1. Create an app using the DynamicSerilogBase package
2. Create a serilog configuration in code setting the minlevel to Information and pass it in to `AddDynamicSerilog()`
3. DO NOT put a serilog section in the appSettings.
4. GET from the loggers endpoint. You'll see that the level is TRACE.

``` cs
var serilogConfig = new LoggerConfiguration()
    .MinimumLevel.Information()
    .WriteTo.Console();
loggingBuilder.AddDynamicSerilog(serilogConfig);
```

## Expected behavior
The log level should report as INFO. 

## Environment (please complete the following information):
 - Platform: On-Prem
 - OS: Windows/Linux
 - .NET Version : Core 3.1 and Net 48
 - Steeltoe Version 3.0.1 and 2.5.1

**Comment by hananiel** [comment]
@jpenniman  the reason we do this is to provide the ability to set log levels at runtime.  We have to take control of the Serilog at the root level as "Verbose" and then filter it down. We cannot filter it "up" as it were. The lines you mark as "Bug" are by design - admittedly, this isn't perfect, but the best we can do considering how Serilog works. So at the moment, we cannot do what you are suggesting.  So you add Serilog and then control verbosity either by config or logger actuator. We can come up with another way to setup configuration in code, probably by passing another method that takes Serilog options via code but that would be an enhancement. We would love to see the PR of a different approach as well.

**Comment by jkonicki** [comment]
Reopening -  There is a solution, but adding a "help wanted" tag in case someone wants to commit a PR for code configuration at a minimum level.

**Comment by jpenniman** [comment]
I or one of my colleagues will consider working on a PR. This is a show stopper for us.

**Comment by hananiel** [comment]
I looked at this again and because of this issue : https://stackoverflow.com/questions/56902169/read-current-serilogs-configuration, there is no way to change the level switch based on current level specified in the Logger configuration. You can however change the level programmatically by specifying SerilogOptions.MinimumLevel in code. 

```csharp
            Host.CreateDefaultBuilder()
             .AddDynamicSerilog((ctx, cfg) =>
                {
                    var config = ctx.Configuration;
                  config["Serilog:MinimumLevel:Default"] = "Information";
                  cfg.ReadFrom.Configuration(ctx.Configuration);
              })
```
If this meets your need, we could create an extension method that lets you Configure SerilogOptions in code, so that you do not have to write this       config["Serilog:MinimumLevel:Default"] = "Information";

**Comment by hananiel** [comment]
Resolved by #718
